pkts-streams/src/main/java/io/pkts/streams/impl/tcpFSM/TcpStreamFSM.java

```diff
@@ -0,0 +1,140 @@
+package io.pkts.streams.impl.tcpFSM;
+
+import io.hektor.fsm.Definition;
+import io.hektor.fsm.FSM;
+import io.hektor.fsm.builder.FSMBuilder;
+import io.hektor.fsm.builder.StateBuilder;
+import io.pkts.packet.TCPPacket;
+import io.pkts.streams.impl.TransportStreamId;
+
+import static io.pkts.streams.impl.tcpFSM.TcpStreamFSM.TcpState.*;
+
+public class TcpStreamFSM{
+
+    public enum TcpState{
+        INIT, HANDSHAKE, ESTABLISHED, FIN_WAIT_1, FIN_WAIT_2,
+        CLOSING_1_CLOSING_2, CLOSED_1_CLOSING_2, CLOSING_1_CLOSED_2, CLOSED
+    }
+
+    public final static Definition<TcpState, TcpStreamContext, TcpStreamData> definition;
+
+    static {
+        final FSMBuilder<TcpState, TcpStreamContext, TcpStreamData> builder=
+                FSM.of(TcpState.class).ofContextType(TcpStreamContext.class).withDataType(TcpStreamData.class);
+
+        builder.withFriendlyName("TCP_stream");
+
+        // define all states for the builder
+        final StateBuilder<TcpState, TcpStreamContext, TcpStreamData> init = builder.withInitialState(INIT);
+        final StateBuilder<TcpState, TcpStreamContext, TcpStreamData> handshake = builder.withState(HANDSHAKE);
+        final StateBuilder<TcpState, TcpStreamContext, TcpStreamData> established = builder.withState(ESTABLISHED);
+        final StateBuilder<TcpState, TcpStreamContext, TcpStreamData> fin_wait_1 = builder.withState(FIN_WAIT_1);
+        final StateBuilder<TcpState, TcpStreamContext, TcpStreamData> fin_wait_2 = builder.withState(FIN_WAIT_2);
+        final StateBuilder<TcpState, TcpStreamContext, TcpStreamData> closing_1_closing_2 = builder.withState(CLOSING_1_CLOSING_2);
+        final StateBuilder<TcpState, TcpStreamContext, TcpStreamData> closed_1_closing_2 = builder.withState(CLOSED_1_CLOSING_2);
+        final StateBuilder<TcpState, TcpStreamContext, TcpStreamData> closing_1_closed_2 = builder.withState(CLOSING_1_CLOSED_2);
+        final StateBuilder<TcpState, TcpStreamContext, TcpStreamData> closed = builder.withFinalState(CLOSED);
+
+        // define all transitions
+        init.transitionTo(HANDSHAKE).onEvent(TCPPacket.class).withGuard(TcpStreamFSM::isSynPacket);
+        init.transitionTo(FIN_WAIT_1).onEvent(TCPPacket.class).withGuard(TcpStreamFSM::isFinPacket);
+        init.transitionTo(CLOSED).onEvent(TCPPacket.class).withGuard(TcpStreamFSM::isRstPacket);
+        init.transitionTo(ESTABLISHED).asDefaultTransition();
```

Comment:
You should avoid default transitions unless you have transient states, which you don't. For a fully functioning FSM, it should be "easy" to determine it's correctness by specifying all valid transitions. Any invalid transitions should error out.

So, I would recommend avoiding the default transition and ensure you specify all cases where it is legal to transition to, in this case, ESTABLISHED from INIT.